backend/api/main.py

```
@@ -0,0 +1,83 @@
+@app.post("/orders")
+def create_order(order: OrderRequest):
+    try:
+        # Validación de inventario
+        try:
+            conn = get_pg_connection()
+            cursor = conn.cursor()
+            for item in order.items:
+                cursor.execute("SELECT stock FROM products WHERE id = %s", (item.id,))
+                result = cursor.fetchone()
+                if not result:
+                    raise HTTPException(status_code=404, detail=f"Producto {item.id} no existe")
```

Ruff (0.13.3)

Abstract `raise` to an inner function

(TRY301)